src/JustEat.StatsD/Buffered/StatsDUtf8Formatter.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace JustEat.StatsD.Buffered
+{
+    internal class StatsDUtf8Formatter
+    {
+        private readonly byte[] _utf8Prefix;
+
+        public StatsDUtf8Formatter(string prefix)
+        {
+            _utf8Prefix = string.IsNullOrWhiteSpace(prefix) ? new byte[0] : Encoding.UTF8.GetBytes(prefix + ".");
+        }
+
+        public int GetMaxBufferSize(in StatsDMessage msg)
+        {
+            const int MaxSerializedDoubleSymbols = 32;
+            const int ColonBytes = 1;
+
+            const int MaxMessageKindSuffixSize = 3;
+            const int MaxSamplingSuffixSize = 2;
+
+            return _utf8Prefix.Length
+                    + Encoding.UTF8.GetByteCount(msg.StatBucket)
```

Comment:
Do the same caching of the encoding into a static field?

Reply:
If it’s consistently slower across several runs, should undo the other one I suggested then.